### PR TITLE
feat(cache): cache control for router and search requests

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|	

Set cache headers that allows browsers and reverse proxy cache

### CODE REVIEW

File: Helper/Cache/CacheHelper.php [hunk 1]
```diff
@@ -53,4 +55,20 @@ public function save(CacheItem $item, array $data)
 
         $this->cache->save($item);
     }
+
+    public static function setCacheHeaders(Request $request, Response $response): Response
+    {
+        if (is_string($response->getContent())) {
+            $response->setCache([
+                'etag' => \md5($response->getContent()),
```
Comment (Davidmattei) [inline Helper/Cache/CacheHelper.php:63]: don't like the hardcoded md5, maybe we should add the hash algo parameter to the commonBundle?

File: Helper/Cache/CacheHelper.php [hunk 2]
```diff
@@ -53,4 +55,20 @@ public function save(CacheItem $item, array $data)
 
         $this->cache->save($item);
     }
+
+    public static function setCacheHeaders(Request $request, Response $response): Response
```
Comment (dameert) [inline Helper/Cache/CacheHelper.php:59]: why are we returning the $response? isn't it passed as an object / modified by reference?
```suggestion
    public static function setCacheHeaders(Request $request, Response $response): void
```

Comment (dameert) [reply Helper/Cache/CacheHelper.php:59]: And the name is "strange" I would expect a "setter" to be setting something in the class that we are currently in. I propose c
```suggestion
    public static function makeResponseCacheable(Request $request, Response $response): void
```

File: Helper/Cache/CacheHelper.php [hunk 3]
```diff
@@ -53,4 +55,20 @@ public function save(CacheItem $item, array $data)
...
+                'max_age' => 600,
+                's_maxage' => 3600,
+                'public' => true,
+                'private' => false,
+            ]);
+            $response->isNotModified($request);
+        }
+
+        return $response;
+    }
```
Comment (dameert) [inline Helper/Cache/CacheHelper.php:73]: And That would make reduce the complexity of the function as follows:

```suggestion
    public static function makeResponseCacheable(Request $request, Response $response): void
    {
        if(! is_string($response->getContent()) {
            return;
        }
        
        $response->setCache([
            'etag' => \md5($response->getContent()),
            'max_age' => 600,
            's_maxage' => 3600,
            'public' => true,
            'private' => false,
        ]);
        $response->isNotModified($request);
    }
```

File: Helper/Cache/CacheHelper.php (hunk 2 above)
Comment (Davidmattei) [reply Helper/Cache/CacheHelper.php:59]: True I aggree

File: Helper/Cache/CacheHelper.php (hunk 3 above)
Comment (Davidmattei) [reply Helper/Cache/CacheHelper.php:73]: 👍

File: Helper/Cache/CacheHelper.php (hunk 1 above)
Comment (theus77) [reply Helper/Cache/CacheHelper.php]: Based on that article I defined sha1 as default etag_hash_algo a0ad9062551337ee00f364f3db4aa75cb693dfc9

File: Helper/Cache/CacheHelper.php (hunk 3 above)
Comment (theus77) [reply Helper/Cache/CacheHelper.php]: c10e0420020120adddea507d70b182cfd0bbd7b0

**Pull request merged**